sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -52,6 +52,7 @@ internal class SessionReceiverManager : ReceiverManager
             {
                 ReceiveMode = sessionProcessor.InnerProcessor.Options.ReceiveMode,
                 PrefetchCount = sessionProcessor.InnerProcessor.Options.PrefetchCount,
+                Identifier = $"S{sessionId}-{sessionProcessor.Identifier}"
```

Comment:
I don't think we can add sessionId here since this could be null for the case where users just want the next available session. Instead we will have to set the Identifier after the ServiceBusSessionReceiver is returned to us on Line 138 based on receiver.SessionId.

Reply:
I'd also suggest we format as: Identifier-S{sessionId}, so that it is easier to match on the beginning of logs.

Reply:
Based on the current pattern the identifier can only be set during creation through the options, because `ServiceBusSessionReceiver.Identifier` is readonly, which is the same as EH, we could do an internal setter on that property or just do something like identifier-sessionreceiver here too if sessionId is null?

Reply:
Good call - I think the internal setter is a good idea.